src/sentry/scim/endpoints/members.py

```diff
@@ -371,10 +375,19 @@ def put(self, request: Request, organization, member):
         if requested_role not in allowed_roles:
             raise SCIMApiError(detail=SCIM_400_INVALID_ORGROLE)
 
+        previous_role = member.role
+        previous_restriction = member.flags["idp:role-restricted"]
         member.role = requested_role
         member.flags["idp:role-restricted"] = idp_role_restricted
         member.save()
 
+        # only update metric if the role changed
+        if (
+            previous_role != organization.default_role
+            or previous_restriction != idp_role_restricted
+        ):
+            metrics.incr("sentry.scim.member.update_role", tags={"organization": organization})
+
```

Comment:
There was an edge case here where if we had:
`current Unrestricted Default role + blank sentryOrgRole -> No Change`

it would still run the metric b/c there was no early return. We fix this by checking that the role or restriction is changed before sending the metric